### ISSUE TITLE
Error in the Example of hyperparameter tuning 

### ISSUE THREAD

**Issue by jpzhangvincent** [scikit-multilearn/scikit-multilearn#61]
I run the test code on the documentation page http://scikit.ml/api/model_estimation.html#model-estimation about hyper-parameter tuning. I got an error from the case of ensemble classifier.
------------ Code ---------------
from skmultilearn.ensemble.rakeld import RakelD
from skmultilearn.problem_transform import BinaryRelevance, LabelPowerset
from sklearn.model_selection import GridSearchCV
from sklearn.naive_bayes import MultinomialNB

x, y = make_multilabel_classification(sparse=True, n_labels=5,
                                      return_indicator='sparse', allow_unlabeled=False)

parameters = {
    'labelset_size': range(2, 3),
    'classifier': [LabelPowerset(), BinaryRelevance()],
    'classifier__classifier': [MultinomialNB()],
    'classifier__classifier__alpha': [0.7, 1.0],
}

clf = GridSearchCV(RakelD(), parameters, scoring='f1_macro')
clf.fit(x, y)
-------------Error -------------------------------
ValueError: Found input variables with inconsistent numbers of samples: [66, 1]

**Comment by ChristianSch** [comment]
Hi,

I couldn't replicate the exception. Could you verify the versions of `sklearn` and `scikit-multilearn`?

```
from skmultilearn.ensemble.rakeld import RakelD
from skmultilearn.problem_transform import BinaryRelevance, LabelPowerset
from sklearn.model_selection import GridSearchCV
from sklearn.naive_bayes import MultinomialNB
from sklearn.datasets import make_multilabel_classification

x, y = make_multilabel_classification(sparse=True, n_labels=5,
                                      return_indicator='sparse',
                                      allow_unlabeled=False)

parameters = {
    'labelset_size': range(2, 3),
    'classifier': [LabelPowerset(), BinaryRelevance()],
    'classifier__classifier': [MultinomialNB()],
    'classifier__classifier__alpha': [0.7, 1.0],
}

clf = GridSearchCV(RakelD(), parameters, scoring='f1_macro')
clf.fit(x, y)
```

```
λ pip show scikit-learn
Name: scikit-learn
Version: 0.18
Summary: A set of python modules for machine learning and data mining
Home-page: http://scikit-learn.org
Author: Andreas Mueller
Author-email: amueller@ais.uni-bonn.de
License: new BSD
Location: /usr/local/lib/python2.7/site-packages
Requires:
```
```
λ pip show scikit-multilearn                                                                                                                11:56:34
Name: scikit-multilearn
Version: 0.0.5
Summary: A set of python modules for multi-label classification
Home-page: http://scikit-multilearn.github.io/
Author: Piotr Szymański
Author-email: niedakh@gmail.com
License: BSD
Location: /usr/local/lib/python2.7/site-packages
Requires:
```

**Comment by roywright** [comment]
I get the same error, with scikit-learn 0.19.0.

**Comment by ChristianSch** [comment]
@roywright Did you use the exact same code as op? If not, could you provide it please so that I can try to reproduce it?

**Comment by roywright** [comment]
Yes, the same code you posted on Aug 2.

**Comment by ChristianSch** [comment]
Can reproduce for `0.19.0` with the same exception.

```
>>> clf.fit(x, y)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/model_selection/_search.py", line 638, in fit
    cv.split(X, y, groups)))
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/externals/joblib/parallel.py", line 779, in __call__
    while self.dispatch_one_batch(iterator):
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/externals/joblib/parallel.py", line 625, in dispatch_one_batch
    self._dispatch(tasks)
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/externals/joblib/parallel.py", line 588, in _dispatch
    job = self._backend.apply_async(batch, callback=cb)
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/externals/joblib/_parallel_backends.py", line 111, in apply_async
    result = ImmediateResult(func)
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/externals/joblib/_parallel_backends.py", line 332, in __init__
    self.results = batch()
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/externals/joblib/parallel.py", line 131, in __call__
    return [func(*args, **kwargs) for func, args, kwargs in self.items]
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/externals/joblib/parallel.py", line 131, in <listcomp>
    return [func(*args, **kwargs) for func, args, kwargs in self.items]
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/model_selection/_validation.py", line 437, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "/home/user/Documents/dev/scikit-multilearn/skmultilearn/problem_transform/br.py", line 81, in fit
    X), self.ensure_output_format(y_subset))
  File "/home/user/Documents/dev/scikit-multilearn/skmultilearn/problem_transform/lp.py", line 68, in fit
    self.transform(y))
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/naive_bayes.py", line 579, in fit
    X, y = check_X_y(X, y, 'csr')
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/utils/validation.py", line 552, in check_X_y
    check_consistent_length(X, y)
  File "/home/user/.virtualenvs/skml3/lib/python3.6/site-packages/sklearn/utils/validation.py", line 173, in check_consistent_length
    " samples: %r" % [int(l) for l in lengths])
ValueError: Found input variables with inconsistent numbers of samples: [66, 1]
```

**Comment by niedakh** [comment]
the unit tests on travis also catch this, will investigate

**Comment by niedakh** [comment]
fixed _with_ 47d22ca4860a6bf3dc754a367b67fc0a04afd8b4 and the general overhaul of underlying clusterers, works in master, although parameters to rakeld have changed
